test/unit/runner.spec.js

```diff
@@ -426,7 +439,44 @@ describe('Runner', function() {
         newRunner.runTest();
       }
       expect(fail, 'to throw', 'allow unhandled errors');
-      done();
+    });
+
+    it('should not allow unhandled errors in hooks to propagate through', function() {
+      suite.beforeEach(function() {
+        throw new Error('this error will not propagate');
+      });
+      var runner = new Runner(suite);
+      try {
+        runner.hook('beforeEach', function() {});
+        expect(true, 'to be', true);
```

Comment:
it's difficult to tell what we're trying to express in these tests.  

if you want to ensure that `runner.hook('beforeEach', function() {})` does not throw, you can do:

```js
expect(function() {
  runner.hook('beforeEach', function() {});
}, 'not to throw');
```

which would replace L449-L454 here.  if you *do* expect that call to throw, of course, remove the "not" :wink:

Reply:
thanks, will update